common/docker/client.go

```diff
@@ -118,11 +118,15 @@ func (c *Client) AddObserver(ob ContainerObserver) error {
 						pending.finish(event.ID)
 						ob.ContainerDestroyed(event.ID)
 					case "network:connect":
-						pending.finish(event.ID)
-						ob.ContainerConnected(event.ID)
+						if containerID, ok := event.Actor.Attributes["container"]; ok {
+							pending.finish(containerID)
```

Comment:
I will remove the calls to `pending.finish(...)`, as this will not stop `ob.ContainerStarted(...)` being called.